### PR TITLE
switching from patched branch to main buildah repo

### PR DESCRIPTION
**Pull request by kpelzel**
https://github.com/hpc/charliecloud/issues/497

Switched over from the hpc patched branch to buildah's main repo.

Note: Buildah v1.10.1 uses OCI spec version 1.0.1-dev so I increased the max oci version in ch-run-oci

### CODE REVIEW

**Comment by rstyd** [comment]
Looks good to me. As far as Travis is concerned, I think we need a newer version of golang to build the newest version of buildah.

**Comment by reidpr** [comment]
Currently we're using Go that's pre-installed by Travis, not from Ubuntu.

File: bin/ch-build [hunk 1]
```diff
@@ -11,7 +11,7 @@ Usage:
 
   $ $(basename "$0") [-b BUILDER] [--builder-info] -t TAG [ARGS ...] CONTEXT
 
-BUILDER is one of: ch-grow docker.
+BUILDER is one of: ch-grow docker buildah.
```
Comment (reidpr) [inline bin/ch-build]: List them in alphabetical order.

File: bin/ch-run-oci [hunk 2]
```diff
@@ -16,7 +16,7 @@ import types
 CH_BIN = os.path.dirname(os.path.abspath(
            inspect.getframeinfo(inspect.currentframe()).filename))
 OCI_VERSION_MIN = "1.0.0"
-OCI_VERSION_MAX = "1.0.0"
+OCI_VERSION_MAX = "1.0.1-dev"
```
Comment (reidpr) [inline bin/ch-run-oci]: What's the reasoning to leave minimum at 1.0.0?

File: doc-src/ch-build_desc.rst [hunk 3]
```diff
@@ -21,8 +21,8 @@ build environment if the builder doesn't do this by default).
 Supported builders, unprivileged:
 
   * :code:`buildah`: Buildah in "rootless" mode with no setuid helpers, using
-    :code:`ch-run` (via :code:`ch-run-oci`) for :code:`RUN` instructions. This
-    currently requires a patched Buildah; see the install instructions.
+    :code:`ch-run` (via :code:`ch-run-oci`) for :code:`RUN` instructions.
+    See the install instructions.
```
Comment (reidpr) [inline doc-src/ch-build_desc.rst]: We'll want to state a minimum Buildah version.

**Pull request merged**